cc_code/src/classifiers/nn-classifier.cc

```diff
@@ -38,22 +38,30 @@ using whatprot::KDTEntry;  // in namespace std for swap
 
 namespace whatprot {
 
-KDTEntry::KDTEntry(SourcedData<DyeTrack, SourceCountHitsList<int>>&& dye_track)
-        : dye_track(move(dye_track)) {
+KDTEntry::KDTEntry(const SequencingModel& seq_model,
+                   SourcedData<DyeTrack, SourceCountHitsList<int>>&& dye_track)
+        : seq_model(&seq_model), dye_track(move(dye_track)) {
     hits = this->dye_track.source.total_hits();
 }
 
 KDTEntry::KDTEntry(KDTEntry&& other)
-        : dye_track(move(other.dye_track)), hits(other.hits) {}
+        : seq_model(other.seq_model),
+          dye_track(move(other.dye_track)),
+          hits(other.hits) {}
 
 KDTEntry& KDTEntry::operator=(KDTEntry&& other) {
+    seq_model = other.seq_model;
     dye_track = move(other.dye_track);
     hits = other.hits;
     return *this;
 }
 
 double KDTEntry::operator[](int i) const {
-    return (double)dye_track.value.counts[i];
+    unsigned int num_channels = dye_track.value.num_channels;
+    unsigned int c = i % num_channels;
+    unsigned int t = i / num_channels;
+    const short* t_counts = &dye_track.value.counts[t * num_channels];
+    return seq_model->channel_models[c]->adjusted_mu(t_counts);
```

Comment:
I went ahead and added a comment, which I should have done from the start. Yes you have the gist of it. I think you meant "if i is 1" not "if i is 2" though, because of 0-indexing.

For adjusted mu. whatprot's implementation allows both a flat and exponential interaction to be set simultaneously for the same pair of channels. This struck me as no harder than having an option to do one or the other, and I figured it could come in handy down the line. They default to 1.0 so that they have no effect (although I just checked and they get subtracted from 1 between the config file and the adjusted_mu() function but have the same kind of parameter name which is idiotic, but it does do as intended).

Reply:
Hmmm - I think I did mean "if `i` is 2" -- I was just picking an arbitrary `i`, in this case the the 3rd entry in the counts array -- which corresponds to `t0c2` in my example array.

So `c` = `i` % `num_channels` ( 2%3) is 2, which is used to select the channel model for the 3rd channel

And `t` = `i` / `num_channels` (2/3) is 0, which is used to select the channel counts at timestep 0.

I did previously have a mistake in my comment -- in which I said `c` would be 1 -- this should have been 2 as edited.